Add tox and workflow to run tests

anarute opened the pull request:
These tools will make the tests and lint checking way easier.

It's possible to run the tests by running `pytest` or `tox` commands. The main difference for now is that with `tox`, it tests in both python 3.8 and 3.9 environments.

Soon I'll also add the lint and coverage stages in the `tox.ini` file, so running `tox` can make all the code checks at once.

Since the tests needed a database, I've added in the settings file a simple SQLite config as a default db.

alex-voodoo commented on ballots/models.py:
```diff
@@ -1,6 +1,6 @@
 from django.db import models
 from django.utils import timezone
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
```
Just wondering, why is this needed?

anarute replied:
`ugettext_lazy` was deprecated: https://docs.djangoproject.com/en/3.2/releases/3.0/#id3

When I run the tests with pytest I saw the warnings

anarute commented on team/settings.py:
```diff
@@ -146,3 +146,10 @@ def set_target_blank(attrs, new=False):
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': 'localdb',
+    }
+}
```
I've restructured the patch adding the `test_settings` file to avoid mixing up with `site_settings`.

I also moved `site_settings` import to the end of `settings` so it can overwrite the default settings.

anarute commented:
@alex-voodoo Sorry for the delay, I have changed the patch a bit and removed pytest for now. I like it better than unittest, I prefer its output, also how it handles fixtures but it's a matter of taste so we can keep the default test library for now and if I miss a specific feature from pytest in the future we can re-discuss switching :)

anarute commented:
And now the tests run on every Pull request and on pushing/merging to `main` :)

I noticed the project requires at least python 3.8 to run without failing, since 3.7 end of support is already [next year](https://www.python.org/downloads/) I'm not sure if it's worth it to make it compatible.

alex-voodoo commented:
I think that we can require Python 3.8.  The project Readme says 3.10.

alex-voodoo commented on team/settings.py:
```diff
@@ -146,3 +144,8 @@ def set_target_blank(attrs, new=False):
+try:
+    from .site_settings import *
+except ImportError:
+    pass
```
Maybe add here some warning, like "default settings are not good for the production"?

anarute replied:
I've added a warning

anarute commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,29 @@
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.8, 3.9]
```
Sounds good, do you know what version is running on the production server?

alex-voodoo replied:
Ohhhh, bad news, there we have 3.7.3.

anarute replied:
That's interesting how it's running without issues! haha ok I'll try to run here with it and check the compatibility

anarute replied:
I've run locally and tested with tox with 3.7 and it has no issue so I'm betting on some specific version of a library that is used in the github action container that is not compatible, so I think it's ok to ignore this for now. I've added 3.10, I would keep 3.8 and 3.9 to test, as they are still used in many places and are compatible with [Django 4.1](https://docs.djangoproject.com/en/4.1/faq/install/#faq-python-version-support) (that I think we should upgrade soon as well :) )

alex-voodoo commented:
:clap:
Let us land it!
